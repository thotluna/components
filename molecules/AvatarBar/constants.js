export const options = {
  sizes: ['sm', 'lg'],
}
