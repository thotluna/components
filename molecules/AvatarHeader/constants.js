export const options = {
  size: ['sm', 'md'],
}
