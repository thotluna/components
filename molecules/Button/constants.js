export const options = {
  colors: ['base', 'primary', 'secondary'],
}
