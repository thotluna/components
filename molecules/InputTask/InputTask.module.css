.inputTask {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  padding: 12px 17px;
  border: 2px solid var(--color-primary);
  background-color: var(--color-background);
  border-radius: var(--border-radius-full);
}
