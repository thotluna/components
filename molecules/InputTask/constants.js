export const status = {
  STATUS_STANDBY: 'standby',
  STATUS_CREATING: 'creating',
}

export const options = {
  status: [status.STATUS_STANDBY, status.STATUS_CREATING],
}
