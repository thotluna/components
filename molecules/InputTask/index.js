export { default } from './InputTask'
export { options } from './constants'
export { default as styles } from './InputTask.module.css'
