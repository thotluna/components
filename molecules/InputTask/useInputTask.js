import { useState, useEffect, useRef } from 'react'
import { status } from './constants'

export default function useInputTask(statusInitial, handelOnCreted) {
  const [statusCurrent, setStatus] = useState(statusInitial)
  const [task, setTask] = useState('')
  const inputRef = useRef()

  const hadelOnEnter = () => {
    handelOnCreted(task)
    setStatus(status.STATUS_STANDBY)
    setTask('')
  }

  const handelCreating = () => {
    setStatus(status.STATUS_CREATING)
  }

  useEffect(() => {
    if (inputRef.current) {
      inputRef.current.focus()
    }
  }, [inputRef.current, statusCurrent])

  return {
    statusCurrent,
    setStatus,
    task,
    setTask,
    hadelOnEnter,
    inputRef,
    handelCreating,
  }
}
