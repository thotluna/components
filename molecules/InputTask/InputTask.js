import React from 'react'
import PropTypes from 'prop-types'
import styles from './InputTask.module.css'
import { options, status } from './constants'
import { getClasses } from '../../helpers/styles'

import Input from '../Input'
import Title from '../../atoms/Title'
import useInputTask from './useInputTask'

export const InputTask = ({ children, statusInitial, handelOnCreted }) => {
  const getStyles = getClasses(styles)({})
  const {
    statusCurrent,
    task,
    setTask,
    hadelOnEnter,
    inputRef,
    handelCreating,
  } = useInputTask(statusInitial, handelOnCreted)

  // const [statusCurrent, setStatus] = useState(statusInitial)
  // const [task, setTask] = useState('')
  // const inputRef = useRef()

  // const hadelOnEnter = () => {
  //   handelOnCreted(task)
  //   setStatus(status.STATUS_STANDBY)
  //   setTask('')
  // }

  // const handelCreating = () => {
  //   setStatus(status.STATUS_CREATING)
  // }

  // useEffect(() => {
  //   if (inputRef.current) {
  //     inputRef.current.focus()
  //   }
  // }, [inputRef.current, statusCurrent])

  return (
    <>
      {statusCurrent === status.STATUS_STANDBY && (
        <div
          className={getStyles('inputTask')}
          tabIndex="0"
          onFocus={handelCreating}
        >
          <Title size="md">{children}</Title>
        </div>
      )}
      {statusCurrent === status.STATUS_CREATING && (
        <Input
          id="input"
          keyEnterMessage="Enter for continued"
          label="input"
          placeholder="is placeholder"
          value={task}
          handelOnChange={(e) => setTask(e.target.value)}
          handelOnKeyPress={hadelOnEnter}
          ref={inputRef}
        />
      )}
    </>
  )
}

InputTask.propTypes = {
  children: PropTypes.string.isRequired,
  statusInitial: PropTypes.oneOf(options.status).isRequired,
  handelOnCreted: PropTypes.func.isRequired,
}

InputTask.defaultProps = {}

export default InputTask
