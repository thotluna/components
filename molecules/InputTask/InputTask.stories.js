import React from 'react'
import InputTask from './InputTask'
import { options, status } from './constants'
import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <InputTask {...args} />
const ListTemplate = getVariusTemplate(InputTask)

export default {
  title: 'Molecules/InputTask',
  inputTask: InputTask,
  args: {
    children: 'What are we going to do today?.',
    statusInitial: status.STATUS_STANDBY,
    handelOnCreted: (title) => {
      console.log(title)
    },
  },
  argTypes: {
    children: { control: 'text' },
    statusInitial: getOptionsArgTypes(options.status),
  },
}

export const Default = Template.bind({})

export const Status = ListTemplate.bind({})
Status.args = {
  items: options.status.map((statusInitial) => ({ statusInitial })),
}
