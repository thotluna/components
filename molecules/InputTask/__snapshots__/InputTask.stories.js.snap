// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/InputTask Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="inputTask"
    onFocus={[Function]}
    tabIndex="0"
  >
    <h3
      className="title color-base"
    >
      What are we going to do today?.
    </h3>
  </div>
</div>
`;

exports[`Storyshots Molecules/InputTask Status 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="inputTask"
    onFocus={[Function]}
    tabIndex="0"
  >
    <h3
      className="title color-base"
    >
      What are we going to do today?.
    </h3>
  </div>
  <div
    className="input-container"
  >
    <div
      className="input-wrap"
    >
      <div
        className="input-frame"
      >
        <input
          aria-label="input"
          className="input"
          id="input"
          name="input"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="is placeholder"
          value=""
        />
        <div
          className="input-message"
        >
          <h4
            className="title color-primary"
          >
            Enter for continued
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
